rollout/service.go

```diff
@@ -277,13 +279,34 @@ func (c *rolloutContext) ensureSVCTargets(svcName string, rs *appsv1.ReplicaSet,
 	}
 	currSelector := svc.Spec.Selector[v1alpha1.DefaultRolloutUniqueLabelKey]
 	desiredSelector := rs.Labels[v1alpha1.DefaultRolloutUniqueLabelKey]
-	if currSelector != desiredSelector {
+	logCtx := c.log.WithField(logutil.ServiceKey, svc.Name)
+
+	if _, ok := svc.Annotations[v1alpha1.ManagedByRolloutsKey]; !ok && currSelector != desiredSelector {
```

Comment:
As we discussed I think this can be refactored a bit to avoid some code duplication and have more clarity by avoiding the need of adding that `else if` condition.